Build failure: rzip

## Steps To Reproduce

1. `nix build github:NixOS/nixpkgs/88195a94#rzip` (this is the current latest nixos-unstable revision)

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
note: keeping build directory '/var/tmp/nix-build-rzip-2.1.drv-2'
error: builder for '/nix/store/a5ksqdgc59lg8iwy2qcfrrfawizpx100-rzip-2.1.drv' failed with exit code 1;
       last 16 log lines:
> Running phase: unpackPhase
> unpacking source archive /nix/store/s0rmi3rmbsskxkr10w44kdlgc01ab54d-rzip-2.1.tar.gz
> source root is rzip-2.1
> setting SOURCE_DATE_EPOCH to timestamp 1139877503 of file rzip-2.1/util.c
> Running phase: patchPhase
> applying patch /nix/store/i05w4awn8qhawx2ba44646z3wh3dizv9-CVE-2017-8364-fill-buffer.patch
> patching file stream.c
> Running phase: updateAutotoolsGnuConfigScriptsPhase
> Running phase: configurePhase
> patching script interpreter paths in ./configure
> ./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
> configure flags: --prefix=/nix/store/ql5phb1p2nlm721h16wr1wf7znjwlld9-rzip-2.1
> creating cache ./config.cache
> checking for gcc... gcc
> checking whether the C compiler (gcc  ) works... no
> configure: error: installation or configuration problem: C compiler cannot create executables.
       For full logs, run 'nix log /nix/store/a5ksqdgc59lg8iwy2qcfrrfawizpx100-rzip-2.1.drv'.
```

</details>

## Additional context

<!-- Add any other context about the problem here. -->

## Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.11-xanmod1, NixOS, 25.05 (Warbler), 25.05.20241213.3566ab7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/frontear/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/frontear/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/frontear/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/frontear/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/smghprxmwyp707kqljlm4fsxm7h5wd72-lix-2.91.1/share`
 - nixpkgs: `/nix/store/haa4yg8jy1pv2qdpai4ps0phyg7qpsnz-input-farm/nixpkgs`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/rz/rzip/package.nix
+++ b/pkgs/by-name/rz/rzip/package.nix
@@ -4,6 +4,7 @@
   fetchurl,
   fetchpatch,
   bzip2,
+  autoreconfHook,
 }:
 
 stdenv.mkDerivation rec {
@@ -14,6 +15,8 @@ stdenv.mkDerivation rec {
     url = "mirror://samba/rzip/rzip-${version}.tar.gz";
     sha256 = "4bb96f4d58ccf16749ed3f836957ce97dbcff3e3ee5fd50266229a48f89815b7";
   };
+
+  nativeBuildInputs = [ autoreconfHook ];
   buildInputs = [ bzip2 ];
 
   patches = [
@@ -23,8 +26,6 @@ stdenv.mkDerivation rec {
       sha256 = "0jcjlx9ksdvxvjyxmyzscx9ar9992iy5icw0sc3n0p09qi4d6x1r";
     })
   ];
-
-  env.NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isClang "-Wno-error=implicit-int";
 
   meta = with lib; {
     homepage = "https://rzip.samba.org/";

